### ISSUE TITLE
udsoncan.MemoryLocation support for 64-bit address space

### ISSUE THREAD

**Issue by driftregion** [pylessard/python-udsoncan#140]
Hi @pylessard , thank you for python-udsoncan.  I have a question about `MemoryLocation`.

https://github.com/pylessard/python-udsoncan/blob/85fdb4e8f2b90adf1b9604b93cab78fa21ff237f/udsoncan/__init__.py#L297 

The specific use-case I'd like to support is making requests to a server with a 64-bit address space, e.g. `address_format=64`. I don't have a copy of the 2006 standard, but Annex H in the 2020 standard seems to indicate that `addressAndLengthFormatIdentifier` > `0x45` is valid, for example: `0x88`

I'm happy to submit a pull request that adds tests and support for `addressAndLengthFormatIdentifier` > `0x45`, but I'd like to first know whether this conflicts with the direction of python-udsoncan.  Specifically:

- are there 2006 clients that rely on `MemoryLocation` throwing a `ValueError` for `addressAndLengthFormatIdentifier` > `0x45` or can this ValueError be safely removed? 
- is the specification of `addressAndLengthFormatIdentifier` different between standard versions? (does `MemoryLocation` need to check `client.config['standard_version']`?)

Thanks!

**Comment by pylessard** [comment]
The standard is not so clear about that honnestly. The examples stops at the values that are supported in the library.

![image](https://user-images.githubusercontent.com/2386969/209677618-5239d79c-c1dc-4071-afe9-a5ca6ca486fb.png)

I think there is no downside to supporting 64 bits addresses. Can you tell me if branch [support-alfid-64bits](https://github.com/pylessard/python-udsoncan/tree/support-alfid-64bits)  is enough for you?

**Comment by driftregion** [comment]
I tried branch [support-alfid-64bits](https://github.com/pylessard/python-udsoncan/tree/support-alfid-64bits).  The `autosize_address` and `autosize_memorysize` functions need updating also.

```sh
  File "/home/user/repos/python-udsoncan/udsoncan/__init__.py", line 375, in __init__
    address_format = self.autosize_address(address)
  File "/home/user/repos/python-udsoncan/udsoncan/__init__.py", line 408, in autosize_address
    raise ValueError("address size must be smaller or equal than 40 bits")
ValueError: address size must be smaller or equal than 40 bits
```

 After updating those functions, it works fine in my case.

[patch.txt](https://github.com/pylessard/python-udsoncan/files/10311264/patch.txt)

**Comment by pylessard** [comment]
Updated. Will merge once you confirm

**Comment by driftregion** [comment]
Confirmed working.

**Comment by pylessard** [comment]
Merged in #142
